Starcounter.Uniform.Tests/Builder/UniDataTableTests.cs

```diff
@@ -0,0 +1,155 @@
+            _dataProviderMock = new Mock<IFilteredDataProvider<Json>>();
+            _dataProviderMock.SetupAllProperties();
+            _returnedRowsFunc = () => new List<Json>();
+            _dataProviderMock.Setup(provider => provider.CurrentPageRows).Returns(() => _returnedRowsFunc());
+            _dataTableColumns = new List<DataTableColumn>();
+            _initialPageSize = 20;
+            _initialPageIndex = 0;
+        }
+
+        private void InitSut()
+        {
+            _sut = new UniDataTable().Init(_dataProviderMock.Object, _dataTableColumns, _initialPageSize, _initialPageIndex);
+        }
+
+        [Test]
+        public void AfterInitSpecifiedColumnShouldBePresent()
+        {
+            var columnDisplayName = "First name";
+            _dataTableColumns.Add(new DataTableColumn { DisplayName = columnDisplayName });
+
+            InitSut();
+
+            _sut.Columns.Should().HaveCount(1);
+            _sut.Columns.Should().ContainSingle().Which.DisplayName.Should().Be(columnDisplayName);
+            _sut.Columns.Should().ContainSingle().Which.LoadRows.Should().NotBeNull();
+        }
+
+        [Test]
+        public void AfterInitInitialPageShouldBeLoaded()
+        {
+            var expectedRow = new Json();
+            var expectedRowsCollection = new List<Json>
+            {
+                expectedRow
+            };
+            var initialPageIndex = 3;
+            _initialPageIndex = initialPageIndex;
+            _dataProviderMock
+                .Setup(provider => provider.CurrentPageRows)
+                .Returns(() =>
+                _dataProviderMock.Object.PaginationConfiguration
+                    .CurrentPageIndex == initialPageIndex
+                    ? expectedRowsCollection
+                    : new List<Json>());
+
+            InitSut();
+
+            _sut.Pages.Should().HaveCount(initialPageIndex + 1);
+            _sut.Pages[initialPageIndex].Rows.Should().ContainSingle().Which.Should().BeSameAs(expectedRow);
+        }
+
+        [Test]
+        public void AfterChangingCurrentPageIndexPaginationConfigurationShouldChange()
+        {
+            var newPageIndex = 2;
+
+            InitSut();
+            _sut.Pagination.Handle(new UniDataTable.PaginationViewModel.Input.CurrentPageIndex { Value = newPageIndex });
+
+            _dataProviderMock.Object.PaginationConfiguration.CurrentPageIndex.Should().Be(newPageIndex);
+            _dataProviderMock.VerifyGet(x => x.CurrentPageRows, Times.Exactly(2));
+        }
+
+        [Test]
+        public void AfterCallingFilterHandleFilterValueInFilterOrderConfigurationShouldChange()
+        {
+            var expectedFilterValue = "John";
+            _dataTableColumns.Add(new DataTableColumn { PropertyName = _columnPropertyName });
+            var filter = new Filter { PropertyName = _columnPropertyName, Value = "Ann" };
+
+            InitSut();
+            _dataProviderMock.Object.FilterOrderConfiguration.Filters.Add(filter);
```

Comment:
"Act" section should be one thing you do. That would be line 99. Everything before that is still "Arrange"

it seems to me you're adding the filter twice. Once directly to the data provider here and again on line 99. Why is it so? If you want to test setting a new filter if another was already present, then write that out as a separate test method.

Reply:
Yes, this test is about checking if calling handle for filter that already exists changes It's value instead of creating new filter again. I'm not sure why and what I should exclude to new test method, since this test is about that.